edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/EdgeletCommandFactory.cs

```diff
@@ -16,13 +16,13 @@ public class EdgeletCommandFactory<T> : ICommandFactory
         readonly ICombinedConfigProvider<T> combinedConfigProvider;
         readonly string edgeDeviceHostname;
         readonly Option<string> parentEdgeHostname;
-        readonly bool checkImagePullBeforeModuleCreate;
+        readonly ModuleUpdateMode moduleUpdateMode;
```

Comment:
this class should not receive moduleUpdateMode, instead the new factory that you added should wrap PrepareUpdateCommand into a BlockingDeploymentCommand (maybe StoppingDeploymentCommand or a different name) and that command should throw the exception.

Reply:
Related to your other comment regarding command factory above. Since there is a lot going on here, will wait for our discussion to clear things up.